libraries/adaptive-expressions/src/expressionProperties/stringExpression.ts

```diff
@@ -27,34 +27,31 @@ export class StringExpression extends ExpressionProperty<string> {
         super(value);
     }
 
-    public tryGetValue(data: object): { value: string; error: Error } {
-        if (typeof this.value == 'string') {
-            let v: any, e: string;
-            const expressionStr = '`' + this.value + '`';
-            ({value: v, error: e} = Expression.parse(expressionStr).tryEvaluate(data));
+    /**
+     * Set a string value.
+     * @param value Value to set.
+     */
+    public setValue(value: string | Expression): void {
+        // reset state to no value or expression
+        super.setValue(undefined);
 
-            return e == undefined ? { value: v as string, error: undefined } : { value: v as string, error: new Error(e) };
+        if (value instanceof Expression) {
+            super.setValue(value);
+            return;
         }
 
-        return super.tryGetValue(data);
-    }
-    
-    public setValue(value: string | Expression): void {
-        this.value = undefined;
-        this.expression = undefined;
-
-        if (typeof value == 'string' && !value.startsWith('=')) {
-            // Trim off the escape char for equals (\=foo) should simply be the string (=foo).
-            if (value.startsWith('\\=')) {
+        if (typeof value == 'string') {
```

Comment:
> want

Got it, btw, do we need to add type check for all kinds of expression properties?

Reply:
I think we should, but I've created a new issue to track it for another milestone: https://github.com/microsoft/botbuilder-js/issues/2568

FYI @luhan2017

Reply:
I've backed out the change so that we can keep the same behavior in the other ExpressionProperties... Thanks for applying the feedback! Let's pick up the discussion in #2568.